modules/builtin/src/content-types/image.js

```diff
@@ -163,16 +164,22 @@ module.exports = {
       return
     }
 
-    let fileName = path.basename(formData.image)
-    if (fileName.includes('-')) {
-      fileName = fileName
-        .split('-')
-        .slice(1)
-        .join('-')
-    }
-    const link = `${formData.BOT_URL}${formData.image}`
+    const link = utils.formatURL(formData.BOT_URL, formData.image)
     const title = formData.title ? ' | ' + formData.title : ''
-    return `Image: [![${formData.title || ''}](<${link}>)](<${link}>) - (${fileName}) ${title}`
+    let fileName = ''
+
+    if (utils.isUrl(link)) {
+      fileName = path.basename(formData.image)
+      if (fileName.includes('-')) {
```

Comment:
I'm not sure which statement you are referring to here since the 'if' at line 173 is the same as before . Can you add some more details?

Reply:
173-178

Reply:
To be honest, I didn't really tried to understand what this block does. I took for granted that I worked properly before, so I simply kept it that way. As you can see they were lines 167-172 before my changes.
 
I simply wrapped it with a condition so that we only format and display pictures that way. Since we now allow expressions, we had to handle formatting everything that is not a URL differently.

I hope this answer your question?